### ISSUE TITLE
8.2.0 ElasticsearchClient search Error with some conditions.

### ISSUE THREAD

**Issue by kimheonseung** [elastic/elasticsearch-java#309]
Hello, I found some strange exception while using ElasticsearchClient - search method.

I'm using
```
Java 1.8.0_202
Elasticsearch Java 8.2.0
Elasticsearch 8.2.0
```

I have index named 'some_index_2022-05' and it has **5,632** documents.  

When I use ElasticsearchClient with some search request, jakarta.json.JsonException occured.

Conditions are like below
* use elasticsearch client with search method when search request's **'from' is over 1,310, Exception occured.**
* use elasticsearch client with search method when search request's **'from' is under 1300, Exception not occured.**

Success and fail case with my code is like below.

```java
// Success with no exception
SearchRequest req = SearchRequest.of(sr -> sr
    .trackTotalHits(h -> h.enabled(Boolean.TRUE))
    .size(10)
    .from(10) // only this point is different with fail case
    .index("some_index_2022-05")
    .sort(st -> st.field( FieldSort.of(f -> f.field("eventTime").order(Order.Desc)));
elasticsearchClient.search(req, MyVO.class);
// and so on...

// Fail with Exception
SearchRequest req = SearchRequest.of(sr -> sr
    .trackTotalHits(h -> h.enabled(Boolean.TRUE))
    .size(10)
    .from(1310) // only this point is different with success case
    .index("some_index_2022-05")
    .sort(st -> st.field( FieldSort.of(f -> f.field("eventTime").order(Order.Desc)));
elasticsearchClient.search(req, MyVO.class);
// and so on...
```

Exception messages are like below.
```shell
(myMethod:123)
 : Error deserializing co.elastic.elasticsearch.core.search.Hit: jakarta.json.JsonException: Jackson exception (JSON path: hits.hits[0]._source) (line no=1, column no=266, offset=-1)

I cannot write more specific stack trace because of company security policy...
```

**But using curl or Chrome Header Extension, same queries are work well.**

meaning
```shell
# success
[POST] /some_index_2022-05/_search
{
  "from": 10,
  "size": 10,
  "sort": [
    {
      "eventTime": {
        "order": "desc"
      }
    }
  ],
  "track_total_hits": true
}

# success
[POST] /some_index_2022-05/_search
{
  "from": 10,
  "size": 10,
  "sort": [
    {
      "eventTime": {
        "order": "desc"
      }
    }
  ],
  "track_total_hits": true
}

```

Is there any options for this with elasticsearch client? 
Please anyone help.
Best regards.

**Comment by swallez** [comment]
Thanks for the very detailed report!

With no details on the stack trace, it's hard to understand what may cause this error. However, the message indicates a nested `Jackson exception`. Can you at least provide the message for this exception?

My best guess at this point would be an error when deserializating the document contained in a `Hit` object into a `MyVO.class`. This could explain why this works with `from: 10` and not with `from: 1310`: a document returned in the 1310-1320 range could be malformed an cause a Jackson deserialization exception.

But we definitely need a bit more information to correctly identify the cause.

**Comment by kimheonseung** [comment]
**"Documents in the 1310-1320 range could be malformed"** make sense..
But every range after that range, same exception occurred.
For example, 1350-1360 would throw the same exception.

For now, I removed index and program logs, changed query using SearchAfter.

But I can reindex same documents.
Maybe I could repeat the same things I mentioned.
In that case, I'll check documents and comment more exception trace.. maybe about next week.
Please do not close this issue until I comment more information about these problems..

Thanks for reply.

**Comment by swallez** [comment]
Can you try replacing `MyV0.class` with `JsonData.class`? This will just load the documents as a JSON node tree. If this works, then this means the issue is with the deserialization of documents into `MyV0` objects.

And yes, I'm keeping this issue open and will wait for more information.

**Comment by kimheonseung** [comment]
Finally, I rechead to the conclusion.
I use `MyVO.class` when I do index.
`MyVO.class` was modified at some point and I forgot about it.
After a few days, I indexed new documents with _**modified** `MyVO.class`_
So, as you expected, documents are not equivalent at some point. (about document at 1310 ~)

For now, I renewed all documents with latest `MyVO.class` and there's no exception with above code.
And I changed `from - to` condition to `search after` condition.
Because I use multiple node and document count is so huge.

Thanks for your help. 
It really helpful to me.

Best regards.
